File: Arrays/Solutions/Construct_Minimum_Bitwise_Array_I.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        ans = []

        for num in nums:
            best = -1

            # Try all possible x from 0 to num
            for x in range(num + 1):
                if (x | (x + 1)) == num:
                    best = x   # smallest valid x
                    break

            ans.append(best)

        return ans

File: Arrays/Solutions/Construct_Minimum_Bitwise_Array_I.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> ans;

        for (int num : nums) {
            int best = -1;

            // Try all possible x from 0 to num
            for (int x = 0; x <= num; x++) {
                // Check the required condition
                if ((x | (x + 1)) == num) {
                    best = x;   // first valid x is the minimum
                    break;
                }
            }

            ans.push_back(best);
        }

        return ans;
    }
};